### PR TITLE
feature/resolution-level-dims

### PR DESCRIPTION
**Pull request by BrianWhitneyAI**
## Description
The purpose of this PR is to resolve https://github.com/bioio-devs/bioio-ome-zarr/issues/19 and add a simple way to access the dimensions of all resolution levels. This PR is a triplet with PRs in [Bioio-Ome-Zarr](https://github.com/bioio-devs/bioio-ome-zarr/pull/23) and [Bioio](https://github.com/bioio-devs/bioio/pull/52).

### CODE REVIEW

File: docs/conf.py
Comment (kmitcham) [inline docs/conf.py:1]: an empty file permission change is an intentional thing?

**Review by kmitcham — APPROVED** [review]

File: docs/conf.py
Comment (BrianWhitneyAI) [reply docs/conf.py:1]: I have no idea why this is happening. Deleting the change locally does not register for git, leaving it as an irrevertable file.

File: bioio_base/reader.py [hunk 1]
```diff
@@ -217,6 +218,26 @@ def current_resolution_level(self) -> int:
...
+            resolution level dictionary of shapes.
+        """
+        if self._resolution_level_dict is None:
+            initial_resoluiton_level = self.current_resolution_level
+            resolution_level_dict = {}
+
+            for level in self.resolution_levels:
+                self.set_resolution_level(level)
+                resolution_level_dict[level] = self.shape
+            self._resolution_level_dict = resolution_level_dict
```
Comment (pgarrison) [inline bioio_base/reader.py:236]: If this is cached in the Reader object, `_reset_self` should delete it for proper interaction with `set_scene`.

File: bioio_base/reader.py [hunk 2]
```diff
@@ -217,6 +218,26 @@ def current_resolution_level(self) -> int:
...
+            self.set_resolution_level(initial_resoluiton_level)
```
Comment (pgarrison) [inline bioio_base/reader.py:237]: Not about your code, but I would like to spread the knowledge of using @property.setter for a setter like `reader.resolution_level = level`. For more, [see here](https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters).

**Review by pgarrison — CHANGES_REQUESTED** [review]

File: bioio_base/reader.py (hunk 1 above)
Comment (toloudis) [reply bioio_base/reader.py:236]: Good catch.  Definitely agree.

**Review by evamaxfield — APPROVED** [review]
Looks good to me. I am just so excited to see this go through ❤️

**Review by SeanLeRoy — APPROVED** [review]

**Review by toloudis — APPROVED** [review]
LGTM! Thank you!

**Review by pgarrison — APPROVED** [review]

**Pull request merged**